### PR TITLE
Expose Android Geocoder isPresent method 

### PR DESCRIPTION
**Pull request by ayouidaniel**
Asume my PR as a full recommendation 🙂

##
## Case of use

You want to ensure that Geocoder implementation exists before executing any code at all. We have a feature where UI changes if Geocoder implementation is available.

### Other changes

Folder and file refactor in order to share and reuse logic and constants.

Android documentation:
- https://developer.android.com/reference/android/location/Geocoder#isPresent()

### CODE REVIEW

File: index.ios.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// @flow
+
+import { NativeModules } from 'react-native';
+import type { Region } from './models/Location';
+import type { Callback } from './models/Callback';
+import debounce from './utils/debounce';
+
+const { RNReverseGeocode } = NativeModules;
+
+const searchForLocations = (
```
Comment (ayouidaniel) [inline index.ios.js]: I decided to keep this method in both files in order to keep platform specific flexibility

File: index.js [hunk 2]
```diff
@@ -1,51 +1,11 @@
-// @flow
+'use strict';
```
Comment (tbergq) [inline index.js]: Why this change?

Comment (tbergq) [reply index.js]: Actually, I don't think this file should be necessary, since having `index.ios` and `index.android`, the metro bundler should automatically choose the correct file

**Comment by tbergq** [comment]
Hi, and thanks a lot for you contribution, we really appreciate it 😄 

I figured we should have our `@kiwicom/eslint-config` added and CI set up before we merge this PR. 

So could you please rebase with master and update your PR?

File: models/Location.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2019-present, [Your company - Change me] All rights reserved.
```
Comment (RobinCsl) [inline models/Location.js]: This ought to be changed, I believe: `[Your company - Change me]`

File: index.ios.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+};
+
+// This method is not available on iOS.
+// We can imply that Geocoding implementation will be always available.
+const isPresent = (
+  callback: Callback,
+  debounceMs: number = 200
+) => {
+  debounce(
+    () => callback(true),
```
Comment (RobinCsl) [inline index.ios.js]: Using this `RNReverseGeocode.isPresent(console.log);` on Android worked like a charm and outputs `true` as expected; however, nothing happens on iOS. Did you manage to have it working on iOS too?

File: index.android.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  debounceMs: number = 200
+) => {
+  debounce(
+    RNReverseGeocode.searchForLocations(searchText, region, callback),
+    debounceMs
+  );
+};
+
+const isPresent = (
+  callback: Callback,
```
Comment (RobinCsl) [inline index.android.js]: I believe the Callback type should be more generic, because the current type is 
```
export type Callback = (err: string, res: Result) => void;
```
and isPresent returning just a boolean does not respect that signature.

File: android/src/main/kotlin/com/kiwi/reversegeocode/RNReverseGeocodeManager.kt [hunk 6]
```diff
@@ -65,6 +65,11 @@ class RNReverseGeocodeManager (reactContext: ReactApplicationContext) : ReactCon
     return formattedAddresses
   }
 
+  @ReactMethod
+  fun isPresent(callback: Callback) {
+    callback.invoke(Geocoder.isPresent())
```
Comment (RobinCsl) [inline android/src/main/kotlin/com/kiwi/reversegeocode/RNReverseGeocodeManager.kt]: To follow the convention for invoking callbacks, could you instead do a try-catch like in the method for searchForLocations?
That means you will have something like `callback.invoke(err, res)` instead of just `callback.invoke(res)`.

File: models/Callback.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+ * @flow
+ */
+
+'use strict';
+
+import type { Address } from './Location';
+
+type Result = $ReadOnlyArray<Address>;
+
+export type Callback = (err: string, res: Result) => void;
```
Comment (RobinCsl) [inline models/Callback.js]: This callback type works for searchForLocations callback; we would need a second Callback type for isPresent, where `res` is a boolean instead.

File: index.js (hunk 2 above)
Comment (RobinCsl) [reply index.js]: One way to achieve this could be:
- remove `index.js`,
- change `"main": "index.js",` in `package.json` to `"main": "index",` and the right file should be picked up.

**Comment by tbergq** [comment]
Closing this due to inactivity
